test/e2e/flowaggregator_test.go

```diff
@@ -102,21 +102,45 @@ const (
 )
 
 func TestFlowAggregator(t *testing.T) {
-	skipIfDualStackCluster(t)
-	data, isIPv6, err := setupTestWithIPFIXCollector(t)
+	data, v4Enabled, v6Enabled, err := setupTestWithIPFIXCollector(t)
 	if err != nil {
 		t.Fatalf("Error when setting up test: %v", err)
 	}
 	defer teardownTest(t, data)
 	defer teardownFlowAggregator(t, data)
 
-	podAIP, podBIP, podCIP, svcB, svcC, err := createPerftestPods(data, isIPv6)
+	podAIP, podBIP, podCIP, err := createPerftestPods(data)
 	if err != nil {
-		t.Fatalf("Error when creating perftest pods and services: %v", err)
+		t.Fatalf("Error when creating perftest pods: %v", err)
 	}
-	// Wait for the Service to be realized.
-	time.Sleep(3 * time.Second)
 
+	if v4Enabled {
+		svcB, svcC, err := createPerftestServices(data, false)
+		if err != nil {
+			t.Fatalf("Error when creating IPv4 perftest services: %v", err)
+		}
+		// Wait for the Service to be realized.
+		time.Sleep(3 * time.Second)
+
+		runTests(t, data, podAIP, podBIP, podCIP, svcB, svcC, false)
+		deletePerftestServices(data)
+	}
+
+	if v6Enabled {
+		svcB, svcC, err := createPerftestServices(data, true)
+		if err != nil {
+			t.Fatalf("Error when creating IPv6 perftest services: %v", err)
+		}
+		// Wait for the Service to be realized.
+		time.Sleep(3 * time.Second)
+
+		runTests(t, data, podAIP, podBIP, podCIP, svcB, svcC, true)
+		deletePerftestServices(data)
+	}
```

Comment:
This looks like the exact same code as above. Maybe define a local function.

```go
testHelper := func(isIPv6 bool) { ... }
if v4Enabled {
    t.Logf("Running tests with IPv4 perftest service")
    testHelper()
}
if v6Enabled
    t.Logf("Running tests with IPv6 perftest service")
    testHelper()
}
```

or even better, use subtests:

```go
testHelper := func(t *testing.T, isIPv6 bool) { ... }
if v4Enabled {
    t.Run("IPv4", func(t *testing.T) { testHelper(t, false) })
}
if v4Enabled {
    t.Run("IPv6", func(t *testing.T) { testHelper(t, true) })
}
```

Reply:
Thanks, addressed it using subtests.